### ISSUE TITLE
How can I use rg in a no live way?

### ISSUE THREAD

**Issue by leiserfg** [camspiers/snap#1]
I just wanna use it as in  `:Rg` of fzf.vim.

**Comment by camspiers** [comment]
You can can provide an `initial_filter`, so you could create a `lua` function:

```
function grep(query)
   snap.run {
     prompt = 'Grep',
     producer = snap.get'producer.ripgrep.vimgrep',
     select = snap.get'select.file'.select,
     multiselect =  snap.get'select.file'.multiselect,
     initial_filter = query
  }
end
```

And then call it from a command you have registered.

**Comment by leiserfg** [comment]
Sorry, I tested it and it does not behave as I asked, is just a live_grep with an initial search term preset.  I was asking for something more like search only once with rg,  fuzzy-search over the list of outputs.
I prefer that kind of search because that way I  can filter over the files' names too.

**Comment by camspiers** [comment]
@leiserfg Ah I see! Sorry for not understanding. While I understand the usecase and think it makes sense, it is probably at the moment out of scope for the project as it doesn't so much pertain to the live searching paradigm of the plugin.

However, I actually use the same approach as you sometimes, and I can show you how I do it.

While this is fennel not lua it should be easy to follow. The following external-grep function prompts for input:

https://github.com/camspiers/dotfiles/blob/4a8ec3a0ebd17d3f79e8bdb27c74464bc961215a/files/.config/nvim/fnl/mappings.fnl#L28-L33

I use which key to bind a mapping here:

https://github.com/camspiers/dotfiles/blob/4a8ec3a0ebd17d3f79e8bdb27c74464bc961215a/files/.config/nvim/fnl/mappings.fnl#L45

And all importantly I set the grepprg for vim in general here such the the inbuilt grep command uses ripgrep:

https://github.com/camspiers/dotfiles/blob/4a8ec3a0ebd17d3f79e8bdb27c74464bc961215a/files/.config/nvim/fnl/options.fnl#L41-L42

**Comment by leiserfg** [comment]
I ended up with chaining  producer-vimgrep and fzy, and it seems to work as I wanted, but still have to test how does it perform in a long codebase
```clojure
  (snap.register.map
    [:n]
    [:<Leader>m]
    (fn []
      (snap.run {:prompt :Grep
                 :initial_filter (vim.fn.input "Grep> ")
                 :producer (fzy (limit 1000 producer-vimgrep))
                 :select select-vimgrep.select
                 :multiselect select-vimgrep.multiselect
                 :views [preview-vimgrep]})))

```

**Comment by camspiers** [comment]
https://github.com/camspiers/snap/commit/dd9e742b9c095a8a615c26b949b7aab00bb87359

I think you will like this feature.

You can now run:

```
  (snap.register.map
    [:n]
    [:<Leader>ff]
    (fn []
      (snap.run {:prompt :Grep
                 :producer (limit 50000 producer-vimgrep)
                 :next {:consumer fzf :config {:prompt :FZF}}
                 :select select-vimgrep.select
                 :multiselect select-vimgrep.multiselect
                 :views [preview-vimgrep]})))
```

And you can change arbitrary config for the next consumer. Then next consumer get provided the current results list from the results view.
